lib/dnsimple/client/vanity_name_servers.rb

```diff
@@ -0,0 +1,43 @@
+module Dnsimple
+  class Client
+    module VanityNameServers
+
+      # Enable vanity name servers for a domain. 
+      #
+      # @see https://developer.dnsimple.com/v1/nameservers/vanity-nameservers/#enable
+      #
+      # @param  [#to_s] domain The domain id or domain name.
+      # @param  [Hash] names A hash of up to 4 external name servers; hash keys
+      # are ns1 through ns4, e.g.
+      #         {
+      #           "ns1" => "ns1.example.com",
+      #           "ns2" => "ns2.example.com"
+      #         }
+      #
+      # @return [void]
+      # @raise  [RequestError] When the request fails.
+      def enable_vanity_name_servers(domain, names)
```

Comment:
I'd imagine we need to add a validation of at least `ns1` being present too, correct? I'm thinking of https://github.com/aetrion/dnsimple-ruby/blob/master/lib/dnsimple/client/domains_records.rb#L33 as an example of this since there is a payload that will be required for this to work properly.

I'm also wondering if it'd be better to have the one using this client provide the `server_source` value, but I don't particularly understand the purpose of `dnsimple` as a value here.